maas/client/viscera/machines.py

```diff
@@ -190,18 +208,107 @@ class Machine(Object, metaclass=MachineType):
         if comment is not None:
             params["comment"] = comment
         data = await self._handler.deploy(**params)
-        return type(self)(data)
+        if not wait:
+            return type(self)(data)
+        else:
+            # Wait for the machine to be fully deployed
+            machine = type(self)(data)
+            while machine.status == NodeStatus.DEPLOYING:
+                await asyncio.sleep(wait_interval)
+                data = await self._handler.read(system_id=self.system_id)
+                machine = type(self)(data)
+            if machine.status == NodeStatus.FAILED_DEPLOYMENT:
+                msg = "{system_id} failed to Deploy.".format(
+                    system_id=machine.system_id
+                )
+                raise FailedDeployment(msg)
+            return machine
 
-    async def release(self, comment: str=None):
+    async def get_power_parameters(self):
+        data = await self._handler.power_parameters(system_id=self.system_id)
+        return data
+
+    async def release(self, comment: str=None, wait: bool=False,
+                      wait_interval: int=5):
         params = {"system_id": self.system_id}
         if comment is not None:
             params["comment"] = comment
         data = await self._handler.release(**params)
-        return type(self)(data)
+        if not wait:
+            return type(self)(data)
+        else:
+            # Wait for machine to be released
+            machine = type(self)(data)
+            while (machine.status == NodeStatus.RELEASING or
+                   machine.status == NodeStatus.DISK_ERASING):
+                await asyncio.sleep(wait_interval)
+                data = await self._handler.read(system_id=self.system_id)
+                machine = type(self)(data)
+            if machine.status == NodeStatus.FAILED_RELEASING:
+                msg = "{system_id} failed to be Released.".format(
+                    system_id=machine.system_id
+                )
+                raise FailedReleasing(msg)
```

Comment:
You are missing FAILED_DISK_ERASING. You should catch that and raise that FailedDiskErasing as well.